core/storage/user/gae_models.py

```diff
@@ -321,3 +321,43 @@ def get_or_create(cls, user_id, collection_id):
             return instance_model
         else:
             return cls.create(user_id, collection_id)
+
+
+class UserQueryModel(base_models.BaseModel):
+    """Model for storing result of queries.
+
+    The id of each instance of this model is alphanumeric id of length 12
+    unique to each model instance.
+    """
+    # Options for a query specified by query submitter.
+    # Query option to specify whether user has created or edited one or more
+    # explorations in last n days.
+    inactive_in_last_n_days = ndb.IntegerProperty(default=None)
+    # Query option to check whether given user has logged in
+    # since last n days.
+    has_not_logged_in_for_n_days = ndb.IntegerProperty(default=None)
+    # Query option to check whether user has created at least
+    # n explorations.
+    created_at_least_n_exps = ndb.IntegerProperty(default=None)
+    # Query option to check whether user has created fewer than
+    # n explorations.
+    created_fewer_than_n_exps = ndb.IntegerProperty(default=None)
+    # Query option to check if user has edited at least n explorations.
+    edited_at_least_n_exps = ndb.IntegerProperty(default=None)
+    # Query option to check if user has edited fewer than n explorations.
+    edited_fewer_than_n_exps = ndb.IntegerProperty(default=None)
+    # List of all user_ids who satisfy all parameters given in above query.
+    # This list will be empty initially. Once query has completed its execution
+    # this list will be populated with all qualifying user ids.
+    user_ids = ndb.StringProperty(repeated=True)
```

Comment:
Hm, I lean towards indexing this as well. It's conceivable that we might use it for a query later on.